### PR TITLE
Simplify and unify on_action signature

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses #941

+ Doc improvements and fixes

### CODE REVIEW

File: src/taipy/gui/gui.py [hunk 1]
```diff
@@ -1238,7 +1237,7 @@ def __get_on_cancel_block_ui(self, callback: t.Optional[str]):
         def _taipy_on_cancel_block_ui(guiApp, id: t.Optional[str], payload: t.Any):
             if _hasscopeattr(self, Gui.__UI_BLOCK_NAME):
                 _setscopeattr(self, Gui.__UI_BLOCK_NAME, False)
-            self.__on_action(id, callback)
+            self.__on_action(id, {"action": callback})
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui.py]: interesting: should self be guiApp here ?

File: src/taipy/gui/gui.py [hunk 2]
```diff
@@ -1636,6 +1635,7 @@ def _hold_actions(
         message: t.Optional[str] = "Work in Progress...",
     ):  # pragma: no cover
         action_name = callback.__name__ if callable(callback) else callback
+        # TODO: what if lambda? (it does work)
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui.py:1638]: it does work as long as there is only one declaration (the other lambda are named by their code :-)

**Review by FredLL-Avaiga — APPROVED** [review]
LGTM
Interesting point: no impact on the FE ...

File: src/taipy/gui/gui.py (hunk 1 above)
Comment (FabienLelaquais) [reply src/taipy/gui/gui.py]: Hmmmm I guess you're right!
Lucky we are that no one is creating multiple Gui instances hu?

File: src/taipy/gui/gui.py (hunk 2 above)
Comment (FabienLelaquais) [reply src/taipy/gui/gui.py:1638]: True.

**Pull request merged**